### PR TITLE
Upgrade Node to v14

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Upgrade to Node v14.18 following on from the [v12 upgrade](https://github.com/guardian/frontend/pull/24344)

## Why?

https://github.com/nodejs/Release

Node v12 is supported until it's EOL of 30 Apr 2022 🌔 
Node v14 is supported until it's EOL of 30 Apr 2023 🌞 

So we can use the latest dependencies with performance, bugfixes and security updates.

## Errors

The following issues were encountered during the upgrade:
- removing as many of the dependency chains that caused Node 14 warning messages on yarn install, primarily:
  ```
  warning chokidar@1.7.0: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
  warning chokidar > fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
  warning fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
  warning core-js@2.6.12: core-js@<3.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
  warning mkdirp@0.5.1: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
  ```
- fix uses of `fs.writeFile` as it now throws an error if the data argument is an invalid type:
  `[ERR_INVALID_ARG_TYPE]: The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView`
- upgrading `mkdirp` brought in the new promisified version so required removed any wrapping with `pify`
- fix unit tests after upgrading `jest-environment-jsdom`
- fix `core-js` import paths after upgrading

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-v12.22.7
+14.18
```
Comment (mxdvl) [inline .nvmrc:1]: I think it’s recommended to pin to a patch version.

Also I prefer it with the `v` prefix 😄

**Review by mxdvl — APPROVED** [review]
I’m loving it.

cc @guardian/client-side-infra for visibility!

File: .nvmrc (hunk 1 above)
Comment (arelra) [reply .nvmrc:1]: I started out with exactly that :)

I then noticed that DCR and the newer repos are only doing minor and dropping the v, so aligned with them.

Makes sense I think to get patch updates - if we trust them not make breaking changes.

Comment (mxdvl) [reply .nvmrc:1]: Someone (@sihil?) mentioned that breaking changes have previously found their way in Node patches… but 🤷

**Review by OllysCoding — APPROVED** [review]
+1

**Review by lucymonie — APPROVED** [review]
I've had a poke around on code and it's looking 👍🏻

**Comment by mchv** [comment]
Nice one @arelra !

**Pull request merged**